### DISCUSSION TITLE
TTS to External Audio

### DISCUSSION THREAD

**Discussion by Aria-Victoria** [karashiiro/TextToTalk#128] [Ideas]
Hello!

Would it be possible to a manual TTS option with selective audio output so we can use TTS to talk in discord from inside the plugin?

I.E. Type text into the plugins text box, press enter, the speech is synthesized, and the audio is output to something like VB audio cable, so it can used in place of a mic for discord and the like.

As someone who uses TTS to speak exclusively, not having to constantly alt tab to use a TTS program would be amazing. (Not to mention actually being able to use polly/azure)

**Comment by karashiiro** [comment]
I'll look into this, I think it's possible based on https://www.markheath.net/post/naudio-audio-output-devices

**Reply by Aria-Victoria** [reply]
I did something similar to this a while ago but unfortunately, I don't have the uncompiled version anymore. I'll attack a zip file with the build and the read me with the text commands below if you have some way of decompiling or poking around it. 

I remember actually getting the device selection to work properly being a pain in the ass but once it was done it worked pretty good. (Had to split the zip files up since it was 33MB and github only takes 25. Sorry. -.-')

[rosettscommands.txt](https://github.com/karashiiro/TextToTalk/files/10071505/rosettscommands.txt)
[TTSRose_fixedP2.7z.zip](https://github.com/karashiiro/TextToTalk/files/10071527/TTSRose_fixedP2.7z.zip)
[TTSRose_fixedP1.7z.zip](https://github.com/karashiiro/TextToTalk/files/10071529/TTSRose_fixedP1.7z.zip)
